cmd/kops-controller/pkg/server/server.go

```diff
@@ -0,0 +1,125 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package server
+
+import (
+	"crypto/tls"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"runtime/debug"
+
+	"github.com/gorilla/mux"
+	"k8s.io/klog"
+	"k8s.io/kops/cmd/kops-controller/pkg/config"
+	"k8s.io/kops/pkg/apis/nodeup"
+	"k8s.io/kops/upup/pkg/fi"
+)
+
+type Server struct {
+	opt      *config.Options
+	server   *http.Server
+	verifier fi.Verifier
+}
+
+func NewServer(opt *config.Options, verifier fi.Verifier) (*Server, error) {
+	server := &http.Server{
+		Addr: opt.Server.Listen,
+		TLSConfig: &tls.Config{
+			MinVersion:               tls.VersionTLS12,
+			PreferServerCipherSuites: true,
+		},
+	}
+
+	s := &Server{
+		opt:      opt,
+		server:   server,
+		verifier: verifier,
+	}
+	r := mux.NewRouter()
+	r.Handle("/bootstrap", http.HandlerFunc(s.bootstrap))
+	server.Handler = recovery(r)
+
+	return s, nil
+}
+
+func (s *Server) Start() error {
+	return s.server.ListenAndServeTLS(s.opt.Server.ServerCertificatePath, s.opt.Server.ServerKeyPath)
+}
+
+func (s *Server) bootstrap(w http.ResponseWriter, r *http.Request) {
+	if r.Body == nil {
```

Comment:
As we control both client & server ... how do you feel about GRPC?

Reply:
I don't see the value of adding the complexity.

Reply:
Fair enough ... it can be helpful in that it effectively takes care of the JSON marshalling/unmarshalling, along with authentication.  But I think it's reasonable to argue that it's just not that hard, and we couldn't use the built-in authentication mechanisms.  (And the really great features come with bigger scale than we're talking about for this API.)